---
title: "Analysis_0.1_Overview_and_rarefaction"
author: "Charles T. Bond"
date: "`r Sys.Date()`"
output: html_document
---

# Microbial Community Analysis: Talladega Forest Non-perennial Stream Network

## Analysis_0.1: Overview and Rarefaction of 16S and ITS ASVs Data

### Introduction
In the previous R markdowns for 16S and ITS bioinformatics (DADA2 pipelines for processing amplicon sequence data), we generated ASV counts and taxonomy for prokaryotes (16S) and fungi (ITS). 

## Setup
```{r, include=FALSE}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE)
setwd("/Users/chunk/Talladega_synoptic")
knitr::opts_knit$set(root.dir = "/Users/chunk/Talladega_synoptic/Analysis")
options(knitr.duplicate.label = "allow") ### so I don't have to give chunks unique names

```

```{r}
library(phyloseq)
library(vegan)
library(ggplot2)
library(tidyr)
library(dplyr)
library(pheatmap)
library(GUniFrac)
library(pals)
library(RColorBrewer)
library(ragg)
library(ggpubr)

#Set theme
theme_set(theme_bw() + theme(
              plot.title = element_text(size=20, color="black"),
              axis.text.x = element_text(size=15, color="black"),
              axis.text.y = element_text(size=15, color="black"),
              axis.title.x = element_text(size=15),
              axis.title.y = element_text(size=15),
              legend.text = element_text(size=12),
              legend.title = element_text(size=15),
            #  legend.position = "bottom",
            #  legend.key=element_blank(),
            #  legend.key.size = unit(0.5, "cm"),
            #  legend.spacing.x = unit(0.1, "cm"),
            #  legend.spacing.y = unit(0.1, "cm"),
              panel.background = element_blank(), 
              #panel.border = element_rect(colour = "black", fill=NA, size=1),
              plot.background = element_blank()))
```

### 16S Data Curation - Phyloseq
```{r}
setwd('/Users/chunk/Talladega_synoptic/Analysis')

#Create phyloseq object
#read in files
asvtab16S <- read.table("16S_04.25.2025.count_table", header=T, row.names=1,
                   check.names=F, sep="\t")

metadata_full <- read.csv("meta_sub_05.12.2025.csv", row.names=1)
metadata_full$substrate <- as.factor(metadata_full$substrate)
## for downstream analyses, we'll also want to concatinate site classification (intermittent and ephemeral) and flow state (Wet versus dry)
metadata_full <- metadata_full %>% 
  mutate(class_state = interaction(int_eph, WetDry, sep = "_"))

### For TAXA, NOTE: We have not yet removed Chloroplasts or Mitochondria. So, we add a filter step
taxtab16S <- read.csv("16S_ASV_taxonomy_04.25.2025.csv", header = T, row.names=1)
taxtab16S<- filter(taxtab16S, Family!='Mitochondria')
taxtab16S<- filter(taxtab16S, Order!='Chloroplast')

asvtab16S <- otu_table(asvtab16S, taxa_are_rows = FALSE)
taxtab16S <- tax_table(as.matrix(taxtab16S))
metadata <- sample_data(metadata_full)
#combine into phyloseq object
pseqtest16S <- phyloseq(asvtab16S, taxtab16S, metadata)
#extract easy-to-use sample data
samdf_16S <- data.frame(sample_data(pseqtest16S))

cured_asvs16S<- pseqtest16S@otu_table
summary(rowSums(cured_asvs16S))
sum(rowSums(cured_asvs16S))
ncol(cured_asvs16S)

### remove samples with less than 1000 reads? We'll wait for rarefaction to trim, let's first just remove zeros. 
#cured_asvs <- cured_asvs[rowSums(cured_asvs)>1000,]
#remove ASVs with zero reads left, we'll do this again aftern rarefying...
#cured_asvs <- cured_asvs[,colSums(cured_asvs)>0]
#summary(rowSums(cured_asvs))

##remove singletons, doubletons, and ASVs with less than 5 reads across all samples. 
cured_asvs16S <- cured_asvs16S[,colSums(cured_asvs16S)>5]
summary(rowSums(cured_asvs16S))

#rowSums(cured_asvs)
pseqtest16S <- phyloseq(cured_asvs16S, taxtab16S, metadata)
#extract easy-to-use sample data
samdf_16S <- data.frame(sample_data(pseqtest16S))

rowSums(cured_asvs16S)

## Examine the read depths by substrate:
###Leaf
summary(rowSums(pseqtest16S@otu_table[pseqtest16S@sam_data$substrate=="L",]))
##L sample n
nrow(pseqtest16S@otu_table[pseqtest16S@sam_data$substrate=="L",])

###Biofilm
summary(rowSums(pseqtest16S@otu_table[pseqtest16S@sam_data$substrate=="B",]))
##B sample n
nrow(pseqtest16S@otu_table[pseqtest16S@sam_data$substrate=="B",])

###Sediment
summary(rowSums(pseqtest16S@otu_table[pseqtest16S@sam_data$substrate=="S",]))
##S sample n
nrow(pseqtest16S@otu_table[pseqtest16S@sam_data$substrate=="S",])

```


### FUNGI - ITS Data Curation - Phyloseq
```{r}
setwd('/Users/chunk/Talladega_synoptic/Analysis')

#Create phyloseq object
#read in files
asvtabITS <- read.table("ITS_04.24.2025.count_table", header=T, row.names=1,
                   check.names=F, sep="\t")
codex<- read.csv("label_codex_TS.csv")

rownames(asvtabITS)<-codex$Sample

metadata_full <- read.csv("meta_sub_05.12.2025.csv", row.names=1)

metadata_full$substrate <- as.factor(metadata_full$substrate)
metadata_full <- metadata_full %>% 
  mutate(class_state = interaction(int_eph, WetDry, sep = "_"))

taxtabITS <- read.csv("ITS_fungi_taxonomy_04.24.2025.csv", header = T, row.names=1)

asvtabITS <- otu_table(asvtabITS, taxa_are_rows = FALSE)
taxtabITS <- tax_table(as.matrix(taxtabITS))
metadata <- sample_data(metadata_full)
#combine into phyloseq object
pseqtestITS <- phyloseq(asvtabITS, taxtabITS, metadata)
#extract easy-to-use sample data
samdftest <- data.frame(sample_data(pseqtestITS))

cured_asvsITS<- pseqtestITS@otu_table
summary(rowSums(cured_asvsITS))
sum(rowSums(cured_asvsITS))
ncol(cured_asvsITS)

### remove samples with less than 1000 reads? We'll wait for rarefaction to trim, let's first just remove zeros. 
#cured_asvs <- cured_asvs[rowSums(cured_asvs)>1000,]
#remove ASVs with zero reads left, we'll do this again aftern rarefying...
#cured_asvs <- cured_asvs[,colSums(cured_asvs)>0]
#summary(rowSums(cured_asvs))

##remove singletons, doubletons, and ASVs with less than 5 reads across all samples. 
cured_asvsITS <- cured_asvsITS[,colSums(cured_asvsITS)>5]
summary(rowSums(cured_asvsITS))
rowSums(cured_asvsITS)


#rowSums(cured_asvs)
pseqtestITS <- phyloseq(cured_asvsITS, taxtabITS, metadata)

## Examine the read depths by substrate:
###Leaf
summary(rowSums(pseqtestITS@otu_table[pseqtestITS@sam_data$substrate=="L",]))
##L sample n
nrow(pseqtestITS@otu_table[pseqtestITS@sam_data$substrate=="L",])

###Biofilm
summary(rowSums(pseqtestITS@otu_table[pseqtestITS@sam_data$substrate=="B",]))
##B sample n
nrow(pseqtestITS@otu_table[pseqtestITS@sam_data$substrate=="B",])

###Sediment
summary(rowSums(pseqtestITS@otu_table[pseqtestITS@sam_data$substrate=="S",]))
##S sample n
nrow(pseqtestITS@otu_table[pseqtestITS@sam_data$substrate=="S",])

### For now, remove samples with less than 1000 reads (arbitrary). We will rarefy later.
pseqtestITS<- prune_samples(sample_sums(pseqtestITS)>=1000, pseqtestITS)
#extract sample data
samdf_ITS <- data.frame(sample_data(pseqtestITS))
summary(rowSums(pseqtestITS@otu_table[#pseqtestITS@sam_data$substrate=="L"
  ,]))
library(stringr)
tax_table(pseqtestITS)[,"Species"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Species"], "s__", ""),`[`, 1)
tax_table(pseqtestITS)[,"Genus"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Genus"], "g__", ""),`[`, 1)
tax_table(pseqtestITS)[,"Family"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Family"], "f__", ""),`[`, 1)
tax_table(pseqtestITS)[,"Order"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Order"], "o__", ""),`[`, 1)
tax_table(pseqtestITS)[,"Class"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Class"], "c__", ""),`[`, 1)
tax_table(pseqtestITS)[,"Phylum"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Phylum"], "p__", ""),`[`, 1)
tax_table(pseqtestITS)[,"Kingdom"] <- sapply(str_replace(tax_table(pseqtestITS)[,"Kingdom"], "k__", ""),`[`, 1)
```
##############################################################################

## ASVs across all samples:
We can visualize the total numbers of ASVs across all samples in two ways:
1) With accumulation curves, showing how many unique ASVs occured across all all sites (by substrate)
2) Venn diagrams shohwing the occurence of
## Accumulation curves from unrarefied 16S data
```{r}

### epilithon leaf sediment 
myColors <- c("#9ACD32","#CD853F","#F4D166")

## I made the first phyloseq object with Fungi only, non-rarefied, because I want to start with this birds'-eye view, overall ASV richness...

botu<- pseqtest16S@otu_table[samdf_16S$substrate=='B',]
lotu<- pseqtest16S@otu_table[samdf_16S$substrate=='L',]
sotu<- pseqtest16S@otu_table[samdf_16S$substrate=='S',]

accum_nor_B <- specaccum(botu, method="random")
accum_nor_L <- specaccum(lotu, method="random")
accum_nor_S <- specaccum(sotu, method="random")

plot(accum_nor_S, col = myColors[3])
#then plot the rest
plot(accum_nor_L, add = TRUE, col = myColors[2]) #col is COLOUR setting, so change it to something else if you want 
plot(accum_nor_B, add = TRUE, col = myColors[1])
plot(accum_nor_S, add = TRUE, col = myColors[3])

legend("topright",y=NULL,legend=as.factor(samdf_16S$substrate),fill=NULL,col = as.factor(samdf_16S$substrate))
#legend("topright",threednmds$xyz.convert(18, 0, 12), pch = 1, col = as.factor(samdftest$substrate), yjust=0, legend = as.factor(samdftest$substrate), cex = 1)

#data <- data.frame(Sites=acc$sites, Richness=acc$richness, SD=acc$sd)
dfb <-data.frame(Sites=accum_nor_B$sites, Richness=accum_nor_B$richness, SD=accum_nor_B$sd)
dfl <-data.frame(Sites=accum_nor_L$sites, Richness=accum_nor_L$richness, SD=accum_nor_L$sd)
dfs <-data.frame(Sites=accum_nor_S$sites, Richness=accum_nor_S$richness, SD=accum_nor_S$sd)

#library(RColorBrewer)
#myColors <- c("#9ACD32","#CD853F","#F4D166")
names(myColors) <- levels(samdf_16S$substrate)
custom_colors <- scale_colour_manual(name = samdf_16S$substrate, values = myColors)

ASVaccumsub16S<- ggplot() +
  geom_point(data=dfl, colour="#CD853F", aes(x=Sites, y=Richness, )) +
geom_line(data=dfl, colour="#CD853F",aes(x=Sites, y=Richness)) +
geom_ribbon(data=dfl,aes(x=Sites,
ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2)+

  geom_point(data=dfb, colour="#9ACD32", aes(x=Sites, y=Richness)) +
geom_line(data=dfb, colour="#9ACD32", aes(x=Sites, y=Richness)) +
geom_ribbon(data=dfb,aes(x=Sites,
ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2)+
  
  geom_point(data=dfs, colour="#F4D166", aes(x=Sites, y=Richness)) +
geom_line(data=dfs, colour="#F4D166", aes(x=Sites, y=Richness)) +
geom_ribbon(data=dfs,aes(x=Sites,
ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2) +
  
  # Color of lines and points
scale_color_discrete(name='Substrate:', labels=c('Epilithic Biofilms', 'Leaf Litter', 'Sediment'))+
  #title(main = "Fungal ASV accumulation curves by substrate")+

                        #, 'Surface #water'))+

#  geom_point(data=dfw, aes(x=Sites, y=Richness, colour='W')) +
#geom_line(data=dfw, aes(x=Sites, y=Richness,colour='W')) +
#geom_ribbon(data=dfw,aes(x=Sites,
#ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2)+
  labs(title = "Prokaryotes (16S)", y="ASV Richness")+
  theme(legend.position="right")
  #scale_fill_discrete(labels=c('High Program',
#  custom_colors
ASVaccumsub16S
#ASVaccumsub16Sb<- ASVaccumsub16S+ theme(legend.position="bottom", plot.margin = unit(c(1, 1, 2, 2), "lines"))
#ASVaccumsub16Sb
plotout <- "16S_ASV_accum.curve_substrate_05.12.25.tiff"
agg_tiff(filename=plotout, width=2555, height=1464, units="px",
         pointsize=10, res=600, compression="lzw", scaling=0.5)
ASVaccumsub16S<- ASVaccumsub16S + scale_color_discrete(name='Substrate:', labels=c('Epilithic Biofilms', 'Leaf Litter', 'Sediment')) + theme(legend.position = "bottom")
ASVaccumsub16S
invisible(dev.off())

save(ASVaccumsub16S, file = "/Users/chunk/Talladega_synoptic/Analysis/Rdata/ASVaccumsub16S.rdata")
load("/Users/chunk/Talladega_synoptic/Analysis/Rdata/ASVaccumsub16S.rdata")
ASVaccumsub16S
```
Saved ASVaccumsub so I can combine it in a Beta Diversity figure with multiple other plots.

## Venn diagram of unrarefied 16S ASVs

```{r}
library(MicEco)
library(eulerr)
#library(RColorBrewer)

myColors <- c("#9ACD32","#CD853F","#F4D166")

venndiagram16S<- ps_venn(pseqtest16S,group="substrate", labels=c('Epilithic biofilms', 'Leaf litter', 'Benthic sediment'), fill=myColors)

venndiagram16S

save(venndiagram16S, file = "/Users/chunk/Talladega_synoptic/Analysis/Rdata/venndiagram16S.rdata")
load("/Users/chunk/Talladega_synoptic/Analysis/Rdata/venndiagram16S.rdata")
venndiagram16S
```

summary stats on taxa before rarefaction 
```{r}
tax_tib16S<-as.data.frame(pseqtest16S@tax_table@.Data)
unique_gen16S<-unique(tax_tib16S$Genus)
###number of genera
length(unique_gen16S)

unique_fam16S<-unique(tax_tib16S$Family)
#number of families
length(unique_fam16S)

unique(tax_tib16S$Phylum)
#unique(tax_tib$Class)
#unique(tax_tib$Family)

## number of ASV without genus assigned
sum(is.na(tax_tib16S$Genus))

## proportion of prokaryote ASVs without genus assigned
sum(is.na(tax_tib16S$Genus))/nrow(tax_tib16S)


```


#############

## Accumulation curves from unrarefied ITS data
```{r}
botu<- pseqtestITS@otu_table[samdf_ITS$substrate=='B',]
lotu<- pseqtestITS@otu_table[samdf_ITS$substrate=='L',]
sotu<- pseqtestITS@otu_table[samdf_ITS$substrate=='S',]

accum_nor_B <- specaccum(botu, method="random")
accum_nor_L <- specaccum(lotu, method="random")
accum_nor_S <- specaccum(sotu, method="random")
myColors <- c("#9ACD32","#CD853F","#F4D166")# ,"#3F8BBA")
plot(accum_nor_S, col = myColors[3])
#then plot the rest
plot(accum_nor_L, add = TRUE, col = myColors[2])
plot(accum_nor_B, add = TRUE, col = myColors[1])
legend("topright",y=NULL,legend=as.factor(samdf_ITS$substrate),fill=NULL,col = as.factor(samdf_ITS$substrate))

dfb <-data.frame(Sites=accum_nor_B$sites, Richness=accum_nor_B$richness, SD=accum_nor_B$sd)
dfl <-data.frame(Sites=accum_nor_L$sites, Richness=accum_nor_L$richness, SD=accum_nor_L$sd)
dfs <-data.frame(Sites=accum_nor_S$sites, Richness=accum_nor_S$richness, SD=accum_nor_S$sd)

library(RColorBrewer)
#myColors <- c("#9ACD32","#CD853F","#F4D166")
names(myColors) <- levels(samdf_ITS$substrate)
custom_colors <- scale_colour_manual(name = samdf_ITS$substrate, values = myColors)

ASVaccumsubITS<- ggplot() +
  geom_point(data=dfl, colour="#CD853F", aes(x=Sites, y=Richness, )) +
geom_line(data=dfl, colour="#CD853F",aes(x=Sites, y=Richness)) +
geom_ribbon(data=dfl,aes(x=Sites,
ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2)+

  geom_point(data=dfb, colour="#9ACD32", aes(x=Sites, y=Richness)) +
geom_line(data=dfb, colour="#9ACD32", aes(x=Sites, y=Richness)) +
geom_ribbon(data=dfb,aes(x=Sites,
ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2)+
  
  geom_point(data=dfs, colour="#F4D166", aes(x=Sites, y=Richness)) +
geom_line(data=dfs, colour="#F4D166", aes(x=Sites, y=Richness)) +
geom_ribbon(data=dfs,aes(x=Sites,
ymin=(Richness-2*SD),ymax=(Richness+2*SD)),alpha=0.2) +

scale_color_discrete(name='Substrate:', labels=c('Epilithic Biofilms', 'Leaf Litter', 'Sediment', 'Water'))+
  labs(title = "Fungi (ITS)", y="ASV Richness")+
  theme(legend.position="right")
  #scale_fill_discrete(labels=c('High Program',
#  custom_colors
ASVaccumsubITS
ASVaccumsubITSb<- ASVaccumsubITS+ theme(legend.position="bottom", plot.margin = unit(c(1, 1, 2, 2), "lines"))
ASVaccumsubITSb
plotout <- "ITS_ASV_accum.curve_substrate_05.12.2025.tiff"
agg_tiff(filename=plotout, width=2555, height=1464, units="px",
         pointsize=10, res=600, compression="lzw", scaling=0.5)
ASVaccumsubITS<- ASVaccumsubITS + scale_color_discrete(name='Substrate:', labels=c('Epilithic Biofilms', 'Leaf Litter', 'Sediment')) + theme(legend.position = "right")
ASVaccumsubITS
invisible(dev.off())

save(ASVaccumsubITS, file = "/Users/chunk/Talladega_synoptic/Analysis/Rdata/ASVaccumsubITS.rdata")
load("/Users/chunk/Talladega_synoptic/Analysis/Rdata/ASVaccumsubITS.rdata")
ASVaccumsubITS

```
Saved ASVaccumsub so I can combine it in a Beta Diversity figure with multiple other plots.

## Venn diagram of unrarefied ITS ASVs

```{r}
library(MicEco)
library(eulerr)
#library(RColorBrewer)

myColors <- c("#9ACD32","#CD853F","#F4D166")#,"#3F8BBA")

venndiagramITS<- ps_venn(pseqtestITS,group="substrate", labels=c('Epilithic biofilms', 'Leaf litter', 'Benthic sediment'), fill=myColors)

venndiagramITS
save(venndiagramITS, file = "/Users/chunk/Talladega_synoptic/Analysis/Rdata/venndiagramITS.rdata")
load("/Users/chunk/Talladega_synoptic/Analysis/Rdata/venndiagramITS.rdata")
venndiagramITS
```



summary stats on taxa before rarefaction 
```{r}
tax_tibITS<-as.data.frame(pseqtestITS@tax_table@.Data)
unique_genITS<-unique(tax_tibITS$Genus)
###number of genera
length(unique_genITS)

unique_famITS<-unique(tax_tibITS$Family)
#number of families
length(unique_famITS)

unique(tax_tibITS$Phylum)
#unique(tax_tib$Class)
#unique(tax_tib$Family)

## number of ASV without genus assigned
sum(is.na(tax_tibITS$Genus))

## proportion without genus assigned
sum(is.na(tax_tibITS$Genus))/nrow(tax_tibITS)

```


## Rarefaction
16S data rarefaction
```{r}
library(GUniFrac)
summary(rowSums(cured_asvs16S))
sum(rowSums(cured_asvs16S))
rowSums(cured_asvs16S)
set.seed(2025)
rarefied16S <- Rarefy(cured_asvs16S, 30550)

rarefied16S$discard
rarasv16S<-rarefied16S$otu.tab.rff
rarasv16S<- rarasv16S[,colSums(rarasv16S)>0]

summary(rowSums(rarasv16S))

## look at read depths for the different substrates:
asvtab16S <- otu_table(rarasv16S, taxa_are_rows = FALSE)
taxtab16S <- tax_table(taxtab16S)

#combine into phyloseq object
pseqtest16S <- phyloseq(asvtab16S, taxtab16S, metadata)

# SAVE RAREFIED 16S PHYLOSEQ object as RDATA
save(pseqtest16S, file = "/Users/chunk/Talladega_synoptic/Analysis/Rdata/pseqtest16S.rdata")
load("/Users/chunk/Talladega_synoptic/Analysis/Rdata/pseqtest16S.rdata")
pseqtest16S
```

## Rarefaction
NMDS of fungi, rarefied data
rarefaction
```{r}
library(GUniFrac)
summary(rowSums(cured_asvsITS))
sum(rowSums(cured_asvsITS))
rowSums(cured_asvsITS)

set.seed(2025)
rarefiedITS <- Rarefy(cured_asvsITS, 1000)

rarefiedITS$discard
rarasvITS<-rarefiedITS$otu.tab.rff
rarasvITS<- rarasvITS[,colSums(rarasvITS)>0]

summary(rowSums(rarasvITS))

## look at read depths for the different substrates:
asvtabITS <- otu_table(rarasvITS, taxa_are_rows = FALSE)
taxtabITS <- tax_table(taxtabITS)

#combine into phyloseq object
pseqtestITS <- phyloseq(asvtabITS, taxtabITS, metadata)

# SAVE RAREFIED 16S PHYLOSEQ object as RDATA
save(pseqtestITS, file = "/Users/chunk/Talladega_synoptic/Analysis/Rdata/pseqtestITS.rdata")
load("/Users/chunk/Talladega_synoptic/Analysis/Rdata/pseqtestITS.rdata")
pseqtestITS
```

Not the rarefied 16S and ITS data, as well as accumulation curves and Venn Diagrams, are exported at .Rdata files. 
This is helpful as my laptop's RAM is a bit overwhelmed handling the full 16S data, so being able to start fresh with just the rarefied data should help things run a little smoother. 




