---
title: "DECIPHER classifier training"
author: "Charles Bond"
date: "04-24-2025"
output: html_document
---

### Classifier Training
This code will train the DECIPHER classifier to classify taxonomy of ITS sequences based on UNITE version 9 for all eukaryotes. Download the general fasta release here to begin the pipeline: https://doi.org/10.15156/BIO/2938070

Note that if you intend to use ensemble tax as we do following our dada2 pipeline (see our amplicon sequence processing bioinformatics pipeline Rmd), you will want to be sure to use the same reference sequence database version (same release of UNITE v9) in the naive bayesian taxonomy assignment step (default in dada2), or else mismatches may be due to database differences. If you want to use a different database, use the same general fasta release to train this classifer and in the dada2 approach. 

### Setup
Note, new users should modify the directory.
```{r setup, include=FALSE}
rm(list = ls())
knitr::opts_chunk$set(echo = TRUE)
setwd("/Users/kuehnlab/Desktop/Bond/DECIPHER_training/Classifier_training")
```


It loads the package before begin or else it gets the error again
```{r}
library("Biostrings")
library(dplyr)
library(stringr)
library(usethis)

library(DECIPHER); packageVersion("DECIPHER")
```
For  DECIPHER package details, see https://doi.org/10.1186/s40168-018-0521-5 

## Training 

Preparing UNITE trainingset!!
Using version sh_general_release_dynamic_s_all_25.07.2023
```{r}
# specify the path to your file of training sequences:
seqs_path <- "sh_general_release_dynamic_s_all_25.07.2023.fasta"
  
  #"utax_uniteref_all_29.11.2022.fasta"
# read the sequences into memory
seqs <- readDNAStringSet(seqs_path)
# Alternatively use readAAStringSet or readRNAStringSet
taxid <- NULL

###Now, they make the assumption that their taxonomies begin with "Root;", but this is not true for the UNITE database. Even if you get through the next step, the `LearnTaxa` function itself demands 'Root;', and therefore we mush insert it before the kingdom level. 
groups <- names(seqs)
groups <- gsub("k__", "Root;k__", groups) 

names(seqs@ranges) <- groups

# obtain the taxonomic assignments
groups <- names(seqs) # sequence names

# assume the taxonomy begins with 'Root;'
groups <- gsub("(.*)(Root;)", "\\2", groups) # extract the group label
groupCounts <- table(groups)
u_groups <- names(groupCounts) # unique groups
length(u_groups) # number of groups

### Pruning the dataset
maxGroupSize <- 10 # max sequences per label (>= 1)
remove <- logical(length(seqs))
for (i in which(groupCounts > maxGroupSize)) {
index <- which(groups==u_groups[i])
keep <- sample(length(index),
maxGroupSize)
remove[index[-keep]] <- TRUE
}
sum(remove) # number of sequences eliminated

```

```{r}
#training the classifier'
maxIterations <- 3 # must be >= 1 to remove putative problem sequences
allowGroupRemoval <- FALSE
probSeqsPrev <- integer() # suspected problem sequences from prior iteration
for (i in seq_len(maxIterations)) {
cat("Training iteration: ", i, "\n", sep="")
# train the classifier
trainingSet <- LearnTaxa(seqs[!remove],
names(seqs)[!remove],
taxid)
# look for problem sequences
probSeqs <- trainingSet$problemSequences$Index
if (length(probSeqs)==0) {
cat("No problem sequences remaining.\n")
break
} else if (length(probSeqs)==length(probSeqsPrev) &&
all(probSeqsPrev==probSeqs)) {
cat("Iterations converged.\n")
break
}
if (i==maxIterations)
break
probSeqsPrev <- probSeqs
# remove any problem sequences
index <- which(!remove)[probSeqs]
remove[index] <- TRUE # remove all problem sequences
if (!allowGroupRemoval) {
# replace any removed groups
missing <- !(u_groups %in% groups[!remove])
missing <- u_groups[missing]
if (length(missing) > 0) {
index <- index[groups[index] %in% missing]
remove[index] <- FALSE # don't remove
}
}
}
sum(remove) # total number of sequences eliminated
length(probSeqs) # number of remaining problem sequences

ranks <- c("root","kingdom", "phylum", "class", "order", "family", "genus", "species") 
trainingSet$ranks <- ranks

### EXPORT THE NEWLY TRAINED MODEL
save(trainingSet, file = "TrainingSet_UNITE_decipher_04.23.2025.RData")

```

